docs/guides/sglang/app.py

```
@@ -0,0 +1,159 @@
+    class GenerateHandler(BaseHTTPRequestHandler):
+        def do_POST(self) -> None:
+            if self.path != "/generate":
+                self.send_error(404)
+                return
+
+            try:
+                length = int(self.headers.get("Content-Length", "0"))
+                payload = json.loads(self.rfile.read(length))
+                prompt = payload["prompt"]
+                if not isinstance(prompt, str) or not prompt.strip():
+                    raise ValueError("prompt must be a non-empty string")
```

Ruff (0.16.2)

[warning] Abstract `raise` to an inner function

(TRY301)